crates/top/rerun/src/commands/rrd/merge_compact.rs

```diff
@@ -205,6 +220,54 @@ fn merge_and_compact(
         }
     }
 
+    let mut pass = 0;
+    while pass < num_passes {
```

Comment:
don't get why this is not a for loop

Reply:
OKAY YOU CAUGHT ME I MIGHT HAVE WRITTEN SOME _NOT RUST_ LATELY